### ISSUE TITLE
Another compilation issue reported as a bug

### ISSUE THREAD

**Issue by mscharley** [cjoudrey/graphql-schema-linter#70]
```
It looks like you may have hit a bugin graphql-schema-linter.

It would be super helpful if you could report this here: https://github.com/cjoudrey/graphql-schema-linter/issues/new

Error: Expected Interface type.
    at invariant (/usr/src/app/node_modules/graphql-schema-linter/node_modules/graphql/jsutils/invariant.js:19:11)
    at produceInterfaceType (/usr/src/app/node_modules/graphql-schema-linter/node_modules/graphql/utilities/buildASTSchema.js:260:83)
    at /usr/src/app/node_modules/graphql-schema-linter/node_modules/graphql/utilities/buildASTSchema.js:334:14
    at Array.map (native)
    at makeImplementedInterfaces (/usr/src/app/node_modules/graphql-schema-linter/node_modules/graphql/utilities/buildASTSchema.js:333:45)
    at interfaces (/usr/src/app/node_modules/graphql-schema-linter/node_modules/graphql/utilities/buildASTSchema.js:312:16)
    at resolveThunk (/usr/src/app/node_modules/graphql-schema-linter/node_modules/graphql/type/definition.js:169:40)
    at defineInterfaces (/usr/src/app/node_modules/graphql-schema-linter/node_modules/graphql/type/definition.js:331:20)
    at GraphQLObjectType.getInterfaces (/usr/src/app/node_modules/graphql-schema-linter/node_modules/graphql/type/definition.js:315:52)
    at typeMapReducer (/usr/src/app/node_modules/graphql-schema-linter/node_modules/graphql/type/schema.js:205:23)
```

```
# test.schema

schema {
  viewer: Viewer
}

type Person {
  email: String
}

# Error triggered because of this.
type Viewer implements Person {
  email: String
}
```

**Comment by cjoudrey** [comment]
😭 so there is a legit issue with the schema `Person` isn't a `interface`, but `graphql-schema-linter` shouldn't be crashing. This is another case of https://github.com/cjoudrey/graphql-schema-linter/issues/53#issuecomment-346191121.

I'll see if I can get an update on https://github.com/graphql/graphql-js/issues/546 as in order to properly fix this crash we need to know when there is syntax errors and what `node` caused it.

In the meantime, we _could_ [rescue `Error` when calling `buildASTSchema`](https://github.com/cjoudrey/graphql-schema-linter/blob/8297952dbce5e791dfad7b0c3120635d9fb64ac0/src/validator.js#L18), but we wouldn't know what file/line caused the issue. We would have to report it to line 0 or something. Definitely a band aid fix, but maybe better than a crash.

**Comment by mscharley** [comment]
Yeah, this case is really annoying specifically because there's *no information at all*. no file, no name, no nothing. I'm not sure of the structure of your code but even if you could return the file instead of crashing that would be great.

**Comment by cjoudrey** [comment]
Yeah, I agree :100:.

The problem is that this error is thrown from `graphql-js` without any information. 😢 Rather than throwing a generic `Error`, `graphql-js` should be throwing some sort of `SyntaxError` that includes the problematic AST `node`. That way `graphql-schema-linter` could report it back properly.

I'll keep you posted. Sorry for the trouble.

**Comment by mscharley** [comment]
No problem! Just reporting things as I run into them. In my case the `Person` type has moved back and forth between an interface and a type a few times while we design the schema and I forgot that we last settled on a type.

**Comment by cjoudrey** [comment]
Will be fixed in next release by #163.
